crates/pathfinder/src/storage/schema/revision_0017.rs

```diff
@@ -0,0 +1,34 @@
+use anyhow::Context;
+use rusqlite::Transaction;
+
+/// Fixes starknet_events table schema.
+///
+/// Our revision 12 migration step had a bug: it didn't upgrade the events schema
+/// in case the events table was empty, leaving us with a database where the
+/// starknet_events table didn't have a stable rowid (via an INTEGER PRIMARY KEY).
+pub(crate) fn migrate(transaction: &Transaction<'_>) -> anyhow::Result<()> {
+    if is_upgrade_required(transaction)? {
+        super::revision_0012::migrate_events_schema(transaction)?
+    }
+
+    Ok(())
+}
+
+fn is_upgrade_required(transaction: &Transaction<'_>) -> anyhow::Result<bool> {
+    let mut stmt = transaction
+        .prepare("SELECT sql FROM sqlite_schema where tbl_name = 'starknet_events'")
```

Comment:
An alternative is to use the column names as in [revision_0002.rs](https://github.com/eqlabs/pathfinder/blob/main/crates/pathfinder/src/storage/schema/revision_0002.rs#L22)